src/query/query_executor.c

```diff
@@ -10892,6 +10896,8 @@ qexec_execute_insert (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xa
   TP_DOMAIN *result_domain;
   bool has_user_format;
 
+  thread_p->no_logging = (bool) insert->no_logging;
```

Comment:
Is it ok to put `no_logging` in `THREAD_ENTRY`? 
I'm concerned with the case in which multiple *_PROC_NODEs are used, especially in the nested way, like nested insert statement:

>create table c (a int) dont_reuse_oid;
create table d (a int, c c);
insert into d (a ,c) values (3, (insert into c (a) values(3)));

This case calls `qexec_execute_insert()` in the nested way, which means can overwrite `thread_p->no_logging` in the inner call.  

Some part of the back trace:

>**#0  qexec_execute_insert (thread_p=0x6dfd30, xasl=0x4e73a18, xasl_state=0x7fffffff7490, skip_aptr=false)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:10871**
#1  0x00007ffff5f5dd38 in qexec_execute_mainblock_internal (thread_p=0x6dfd30, xasl=0x4e73a18, xasl_state=0x7fffffff7490, p_class_instance_lock_info=0x0)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:13754
#2  0x00007ffff5f5d688 in qexec_execute_mainblock (thread_p=0x6dfd30, xasl=0x4e73a18, xstate=0x7fffffff7490, p_class_instance_lock_info=0x0)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:13539
#3  0x00007ffff5f0c814 in fetch_peek_dbval (thread_p=0x6dfd30, regu_var=0x4e739b0, vd=0x7fffffff7490, class_oid=0x7fffffff6ee8, obj_oid=0x0, tpl=0x0, 
    peek_dbval=0x7fffffff6ef8) at /home/jaeeunkim/workspace/cubrid_TDE/src/query/fetch.c:3872
**#4  0x00007ffff5f5939e in qexec_execute_insert (thread_p=0x6dfd30, xasl=0x4e73498, xasl_state=0x7fffffff7490, skip_aptr=false)**
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:11473
#5  0x00007ffff5f5dd38 in qexec_execute_mainblock_internal (thread_p=0x6dfd30, xasl=0x4e73498, xasl_state=0x7fffffff7490, p_class_instance_lock_info=0x0)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:13754
#6  0x00007ffff5f5d688 in qexec_execute_mainblock (thread_p=0x6dfd30, xasl=0x4e73498, xstate=0x7fffffff7490, p_class_instance_lock_info=0x0)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:13539
#7  0x00007ffff5f5ffaa in qexec_execute_query (thread_p=0x6dfd30, xasl=0x4e73498, dbval_cnt=1, dbval_ptr=0x6cf388, query_id=27)
    at /home/jaeeunkim/workspace/cubrid_TDE/src/query/query_executor.c:14817

Reply:
And, is there no need to initialize `thread_p->no_logging` after a query terminates? What happens if thread_p attaches another query with no_logging=true until calling qexec_execute_update/insert/delete? Is there no heap manipulation like updating statistics, catalog, and root class?

Reply:
@wrlawodms fixed them all.